API/hermes/hermes.h

```diff
@@ -188,12 +188,16 @@ class HERMES_EXPORT HermesRuntime : public jsi::Runtime {
   /// Unregister this runtime for sampling profiler.
   void unregisterForProfiling();
 
-  /// Register this runtime for execution time limit monitoring, with a time
-  /// limit of \p timeoutInMs milliseconds.
+  /// Asynchronously terminates the current execution.
   /// All JS compiled to bytecode via prepareJS, or evaluateJS, will support the
   /// time limit monitoring.  If JS prepared in other ways is executed, care
   /// must be taken to ensure that it is compiled in a mode that supports the
   /// monitoring (i.e., the emitted code contains async break checks).
+  void notifyNeedsTermination();
```

Comment:
I'd suggest renaming this to `asyncTriggerTimeout`, to make it clear that this will throw a timeout exception.